db-client/src/index.ts

```diff
@@ -0,0 +1,160 @@
+import { DBClient, Options } from '@binaris/shift-interfaces-node-client';
+import {
+  ClientContext, UpdateOptions, Version, VersionedMaybeObject
+} from '@binaris/shift-interfaces-node-client/interfaces';
+import deepFreeze, { DeepReadonly } from 'deep-freeze';
+
+export interface Versioned<T> {
+  version: Version;
+  value: T;
+}
+
+function serverUrlFor(endpoint: string): string {
+  return `https://${endpoint}/v1`;
+}
+
+const defaultOptions: Options = {
+  timeoutMs: 2000,
+};
+
+// Not an async iterator so as to separate *generation* of the next
+// delay from its *activation*.
```

Comment:
Note that the delay is activated immediately, it is only awaited at a random time in the future.